lib/std/mem.zig

```diff
@@ -83,6 +83,14 @@ pub const Allocator = struct {
         return &slice[0];
     }
 
+    /// Returns a pointer to allocated memory with the provided value.
+    /// Call `destroy` with the result to free the memory.
+    pub fn duplicate(self: *Allocator, value: var) Error!*@TypeOf(value) {
```

Comment:
We already have `.dupe` and `.dupeZ`.

Reply:
@daurnimator  dupe* returns arrays :)

@pixelherodev I propose rename it to `copy`.

Reply:
@data-man Actually, dupe returns a _slice_ ;)

Renaming it is a good idea.

Actually, this makes more sense in `std.mem` also. `std.mem.dupeOne` for instance. The allocator interface should be as small as possible.

Reply:
... or not. Hadn't realized dupe / dupeZ were in the interface now.